lib/src/services/globalConfig.ts

```diff
@@ -0,0 +1,56 @@
+import appContext from "../utils/appContext";
+import fs from "fs";
+import yaml from "js-yaml";
+import { z } from "zod";
+import { createFileIfMissingSync } from "../utils/fileSystem";
+
+const ZGlobalConfigYAML = z.record(
+  z.string(),
+  z.array(
+    z.object({
+      token: z.string(),
+    })
+  )
+);
+
+type GlobalConfigYAML = z.infer<typeof ZGlobalConfigYAML>;
+
+/**
+ * Read data from a global config file
+ * @param file defaults to `CONFIG_FILE` defined in `constants.js`
+ * @param defaultData defaults to `{}`
+ * @returns
+ */
+export function readGlobalConfigData(
+  file = appContext.configFile,
+  defaultData = {}
+): GlobalConfigYAML {
+  createFileIfMissingSync(file);
+  const fileContents = fs.readFileSync(file, "utf8");
+  const yamlData = yaml.load(fileContents);
+  const parsedYAML = ZGlobalConfigYAML.safeParse(yamlData);
+  if (parsedYAML.success) {
+    return parsedYAML.data;
+  }
+  return defaultData;
+}
+
+function writeGlobalConfigData(file: string, data: object) {
+  createFileIfMissingSync(file);
+  const existingData = readGlobalConfigData(file);
+  const yamlStr = yaml.dump({ ...existingData, ...data });
+  fs.writeFileSync(file, yamlStr, "utf8");
+}
+
+/**
+ * Save a token to the global config file
+ * @param file The path to the global config file
+ * @param hostname The hostname to save the token for
+ * @param token The token to save
+ */
+export function saveToken(file: string, hostname: string, token: string) {
+  const data = readGlobalConfigData(file);
+  data[hostname] = []; // only allow one token per host
+  data[hostname][0] = { token };
```

Comment:
Nit: You can do this on one line, which I think is a bit more readable.
```suggestion
  data[hostname] = [{ token }]; // only allow one token per host
```